### PR TITLE
feat(card): add card component

### PR DESCRIPTION
**Pull request by vikram-raj**
Added a card component. This is a dynamic component and it is highly configurable. Example for this component is also added in this PR.

### CODE REVIEW

File: src/app/card/card.model.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  /**
+   * Should be number with unit.
+   * This should be shorthand format
+   * Default value 0 0 0 0
+   */
+  margin?: string;
+}
+
+export let defaultCardConfig = {
```
Comment (sanbornsen) [inline src/app/card/card.model.ts]: Make it constant

**Comment by dlabrecq** [comment]
Pinging @mindreeper2420 as Adam is our expert on cards.

File: src/app/card/examples/card-example.component.less [hunk 2]
```diff
@@ -0,0 +1,28 @@
+@import (reference) '../../../assets/stylesheets/ngx-widgets.less';
+
+.card-header {
+  .creator {
+    color: #bbb;
+  }
+}
+.card-footer {
+  .label {
+    color: #000;
```
Comment (mindreeper2420) [inline src/app/card/examples/card-example.component.less]: use color variables, not hex codes

File: src/app/card/examples/card-example.component.less [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    background-color: #a18fff;
```
Comment (mindreeper2420) [inline src/app/card/examples/card-example.component.less]: same here

File: src/app/card/examples/card-example.component.less [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    display: inline-block;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    overflow: hidden;
+    vertical-align: bottom;
+    border-radius: 2px;
+    box-shadow: 1px 1px 1px rgba(80, 80, 80, .4);
+    font-size: 12px;
```
Comment (mindreeper2420) [inline src/app/card/examples/card-example.component.less]: should be a relational size to the base font (`em` or `rem).

**Comment by mindreeper2420** [comment]
Are the styles for this component solely based on PatternFly, or do they also include the custom card styles that are in fabric8-ui? We created the custom styles, as the dashboards cards do not have a representative view in PatternFly and were deemed too customized for OSIO to be a common PatternFly component.

**Comment by vikram-raj** [comment]
@mindreeper2420 This card component is a dynamic component we can add any styles as per requirement. We are going to use this card component in planner board. [Here](https://redhat.invisionapp.com/share/TYLU3G8JBE3#/screens/305555707) is link of the board UX.

**Comment by dlabrecq** [comment]
Can this component achieve the same layout as what is used for the fabric8-ui home and space dashboards? If so, an example of that would be helpful.

If the layout is specific to Planner (e.g., the header/borders seem very different than what's used in fabric8-u), perhaps the component should be named board-view-card so it's clear this is not a standard card layout?

**Comment by mindreeper2420** [comment]
@vikram-raj IMO, if a card component is being created outside of the ones in PatternFly-ng, it should be able to handle all of the custom cards needed for OSIO, not just Planner.

Is there a particular reason why Planner needs this custom one? From the design, it looks like a standard card component, as found in PatternFly-ng (as it has a title and body content). 

The [Trend Card](http://www.patternfly.org/pattern-library/cards/trend-card/#overview) from PatternFly 3 has a footer in it, so that is already included with the PatternFly library.

**Comment by joshuawilson** [comment]
I agree with @mindreeper2420, this repo is for shared and shareable components. If something is locked into Planner then keep it there. 

@vikram-raj have you looked at what @dlabrecq said?

**Comment by vikram-raj** [comment]
Yes, @joshuawilson I looked into patternfly-ng card component. It is good but we can improve patternfly-ng card component providing more configuration.

> Can this component achieve the same layout as what is used for the fabric8-ui home and space dashboards?

Yes @dlabrecq we can achieve the same layout using this component as what is used for the fabric8-ui.

> If the layout is specific to Planner (e.g., the header/borders seem very different than what's used in fabric8-u), perhaps the component should be named board-view-card so it's clear this is not a standard card layout?

It is not specific to planner we can use it anywhere. I used ```ng-template``` for header, body and footer so we can put any template into it as per requirment. Border, shadow, boarder radius all these options are optional and configurable.

**Comment by mindreeper2420** [comment]
Why not contribute to PatternFly-ng then?

**Comment by vikram-raj** [comment]
@mindreeper2420 Yeah, I will do.

**Comment by joshuawilson** [comment]
Can we close this then?

**Comment by joshuawilson** [comment]
Closing this as you said you will do this in PF.

**Pull request closed without merging**